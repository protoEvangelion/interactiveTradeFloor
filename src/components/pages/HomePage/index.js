import React, { Component } from 'react'

import { PageTemplate, Header } from 'components'

import { Hero } from 'containers'

class HomePage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      filter: 'None',
    }
    this.filterOwner = this.filterOwner.bind(this)
  }
  filterOwner(filter) {
    this.setState({ filter })
  }
  render() {
    return (
      <PageTemplate header={<Header filter={this.filterOwner} />} hero={<Hero filter={this.state.filter} />} />
    )
  }
}

export default HomePage
