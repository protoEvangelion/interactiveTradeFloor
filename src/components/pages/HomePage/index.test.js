import React from 'react'
import sinon from 'sinon'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import HomePage from '.'

describe('<HomePage />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <HomePage />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
