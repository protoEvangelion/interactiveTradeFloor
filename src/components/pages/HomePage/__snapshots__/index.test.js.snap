// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders correctly 1`] = `<span />`;
